hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/client/HddsClientUtils.java

```diff
@@ -70,6 +71,9 @@ private HddsClientUtils() {
           // Not Replicated Exception will be thrown if watch For commit
           // does not succeed
           .add(NotReplicatedException.class)
+          // for reconPipelineReportHandler , if could not find pipeline
```

Comment:
And also I see this list is being used in KeyOutputStream and XceiverClientRatis.
Not understood why we need this to be added here.

Reply:
when a scmclient  is taking a grpc call, if any exception in this list thrown,  it will stop retry, or else it will try for a maxcount at a specified interval , which are set at `SCMClientConfig.java`. here , i want `scmclient.getpipeline()` not to retry if PipelineNotFoundException is threw.

Reply:
And also we don't use this, we use the list specified in 
```
  private static final List<Class<? extends Exception>>
      NON_RETRIABLE_EXCEPTION_LIST =
      ImmutableList.<Class<? extends Exception>>builder()
          .add(SCMException.class)
          .add(NonRetriableException.class)
          .build();
```
SCMHAUtils.java

And also you can refer SCMHAUtils#getRetryAction for this.

Reply:
do you mean if i want to forbid this retry here for pipelineNotFoundException , i should set it here?